Sources/Rendering/Core/Picker/index.js

```diff
@@ -89,9 +89,171 @@ function vtkPicker(publicAPI, model) {
+        //
+        //  Get the bounding box of the modeller.  Note that the tolerance is
+        //  added to the bounding box to make sure things on the edge of the
+        //  bounding box are picked correctly.
+        const bounds = mapper
+          ? vtkBoundingBox.inflate(mapper.getBounds(), tolerance)
+          : [0, 0, 0, 0, 0, 0];
+
+        if (
+          vtkBoundingBox.intersectBox(bounds, p1Mapper, ray, hitPosition, t)
+        ) {
+          t[0] = publicAPI.intersectWithLine(
+            p1Mapper,
+            p2Mapper,
+            tolerance * 0.333 * (scale[0] + scale[1] + scale[2]),
+            prop,
+            mapper
+          );
+
+          if (t[0] < Number.MAX_VALUE) {
+            const p = [];
+            p[0] = (1.0 - t[0]) * p1World[0] + t[0] * p2World[0];
+            p[1] = (1.0 - t[0]) * p1World[1] + t[0] * p2World[1];
+            p[2] = (1.0 - t[0]) * p1World[2] + t[0] * p2World[2];
+
+            let actorId = -1;
+            for (let i = 0; i < model.actors.length; i++) {
+              if (model.actors[i] === prop) {
+                actorId = i;
+                break;
+              }
+            }
+
+            if (actorId !== -1) {
+              // If already in list, set point to the closest point
+              const oldPoint = model.pickedPositions[actorId];
+              if (
+                vtkMath.distance2BetweenPoints(p1World, p) <
+                vtkMath.distance2BetweenPoints(p1World, oldPoint)
+              ) {
+                model.pickedPositions[actorId] = p;
+              }
+            } else {
+              model.actors.push(prop);
+              model.pickedPositions.push(p);
+            }
+          }
+        } // if visible and pickable and not transparent
+      }
+    });
+    // sort array by distance
+    const tempArray = [];
+    for (let i = 0; i < model.pickedPositions.length; i++) {
+      tempArray.push({
+        actor: model.actors[i],
+        pickedPosition: model.pickedPositions[i],
+        distance2: vtkMath.distance2BetweenPoints(
+          p1World,
+          model.pickedPositions[i]
+        ),
+      });
+    }
+    tempArray.sort((a, b) => {
+      const keyA = a.distance2;
+      const keyB = b.distance2;
+      // order the actors based on the distance2 attribute, so the near actors comes
+      // first in the list
+      if (keyA < keyB) return -1;
+      if (keyA > keyB) return 1;
+      return 0;
+    });
+    model.pickedPositions = [];
+    model.actors = [];
+    tempArray.forEach((obj) => {
+      model.pickedPositions.push(obj.pickedPosition);
+      model.actors.push(obj.actor);
+    });
+  }
+
   // Intersect data with specified ray.
   // Project the center point of the mapper onto the ray and determine its parametric value
-  publicAPI.intersectWithLine = (p1, p2, tol, mapper) => {
+  publicAPI.intersectWithLine = (p1, p2, tolerance, prop, mapper) => {
```

Comment:
This changes the public API for the vtkPicker, which should be reflected in the vtkPicker types.

Reply:
Yep, will do. 
I need to update it to comply with `CellPicker.intersectWithLine`.